compiler/src/iree/compiler/Dialect/VM/Target/Bytecode/BytecodeModuleTarget.cpp

```diff
@@ -460,6 +460,15 @@ static LogicalResult buildFlatBufferModule(
         return iree_vm_ImportFunctionDef_end(fbb);
       }));
 
+  // Check that all export ops point to valid functions.
+  for (auto &exportFuncOp : exportFuncOps) {
+    if (!symbolTable.lookup<IREE::VM::FuncOp>(exportFuncOp.function_ref())) {
+      return moduleOp.emitError()
+             << "vm.func op named '" << exportFuncOp.function_ref()
+             << "' not found for export";
```

Comment:
I tried to add a test for this, but `iree-compile` does not recognize the `--verify-diagnostics` flag needed for `// expected-error` tests (such as in https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Dialect/VM/Target/Bytecode/test/module_encoding_smoke.mlir).